operator-assignment fix is breaking the code

**Tell us about your environment**

* **ESLint Version:** v3.18.0
* **Node Version:** v7.7.2
* **npm Version:** v4.1.2

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```js
module.exports = {
    root: true,
    parserOptions: {
        ecmaVersion: 8,
    },
    rules: {
        'operator-assignment': [2, 'always'],
    },
};
```

**What did you do? Please include the actual source code causing the issue.**
#### test.js
```js
tmp = tmp + (2 * options.x[y]);
```

#### `$ eslint test.js --fix`

**What did you expect to happen?**
I expected the fix function to fix the code.

**What actually happened? Please include the actual, raw output from ESLint.**
Instead, it broke the code.
#### `$ eslint test.js --fix > output.txt`
```js
c:\nodejs\test\test.js
  1:25  error  Parsing error: Unexpected token ;

✖ 1 problem (1 error, 0 warnings)
```

#### test.js
```js
tmp += (2 * options.x[y];
```

It seems, that the rule doesn't handle braces properly.

--- a/lib/rules/operator-assignment.js
+++ b/lib/rules/operator-assignment.js
@@ -135,7 +135,7 @@ module.exports = {
                                 const equalsToken = getOperatorToken(node);
                                 const operatorToken = getOperatorToken(expr);
                                 const leftText = sourceCode.getText().slice(node.range[0], equalsToken.range[0]);
-                                const rightText = sourceCode.getText().slice(operatorToken.range[1], expr.right.range[1]);
+                                const rightText = sourceCode.getText().slice(operatorToken.range[1], node.right.range[1]);
 
                                 return fixer.replaceText(node, `${leftText}${expr.operator}=${rightText}`);
                             }

--- a/tests/lib/rules/operator-assignment.js
+++ b/tests/lib/rules/operator-assignment.js
@@ -155,6 +155,16 @@ ruleTester.run("operator-assignment", rule, {
         options: ["always"],
         errors: EXPECTED_OPERATOR_ASSIGNMENT
     }, {
+        code: "x = x + (y)",
+        output: "x += (y)",
+        options: ["always"],
+        errors: EXPECTED_OPERATOR_ASSIGNMENT
+    }, {
+        code: "x += (y)",
+        output: "x = x + (y)",
+        options: ["never"],
+        errors: UNEXPECTED_OPERATOR_ASSIGNMENT
+    }, {
         code: "x += y",
         output: "x = x + y",
         options: ["never"],

